docs/standard/native-interop/best-practices.md

```diff
@@ -0,0 +1,245 @@
+
+
+The following types, being pointers, do follow the width of the platform. Use `IntPtr`/`UIntPtr` for these.
+
+| Signed Pointer Types (use `IntPtr`) | Unsigned Pointer Types (use `UIntPtr`) |
+|:------------------------------------|:---------------------------------------|
+| `HANDLE`                            | `WPARAM`                               |
+| `HWND`                              | `UINT_PTR`                             |
+| `HINSTANCE`                         | `ULONG_PTR`                            |
+| `LPARAM`                            | `SIZE_T`                               |
+| `LRESULT`                           |                                        |
+| `LONG_PTR`                          |                                        |
+| `INT_PTR`                           |                                        |
+
+A Windows `PVOID` which is a C `void*` can be marshaled as either `IntPtr` or `UIntPtr`, but prefer `void*` when possible.
+
+[Windows Data Types](/windows/desktop/WinProg/windows-data-types)
+
+[Data Type Ranges](/cpp/cpp/data-type-ranges)
+
+## Structs
+
+Managed structs are created on the stack and aren't removed until the method returns. By definition then, they are "pinned" (it won't get moved by the GC). You can also simply take the address in unsafe code blocks if native code won't use the pointer past the end of the current method.
+
+Blittable structs are much more performant as they can simply be used directly by the marshalling layer. Try to make structs blittable (for example, avoid `bool`). For more information, see the [Blittable Types](#blittable-types) section.
+
+*If* the struct is blittable, use `sizeof()` instead of `Marshal.SizeOf<MyStruct>()` for better performance. As mentioned above, you can validate that the type is blittable by attempting to create a pinned `GCHandle`. If the type is not a string or considered blittable, `GCHandle.Alloc` will throw an `ArgumentException`.
+
+Pointers to structs in definitions must either be passed by `ref` or use `unsafe` and `*`.
+
+**✔️ DO** match the managed struct as closely as possible to the shape and names that are used in the official platform documentation or header.
+
+**✔️ DO** use the C# `sizeof()` instead of `Marshal.SizeOf<MyStruct>()` for blittable structures to improve performance.
+
+An array like `INT_PTR Reserved1[2]` has to be marshaled to two `IntPtr` fields, `Reserved1a` and `Reserved1b`. When the native array is a primitive type, we can use the `fixed` keyword to write it a little more cleanly. For example, `SYSTEM_PROCESS_INFORMATION` looks like this in the native header:
+
+```c
+typedef struct _SYSTEM_PROCESS_INFORMATION {
+    ULONG NextEntryOffset;
+    ULONG NumberOfThreads;
+    BYTE Reserved1[48];
+    UNICODE_STRING ImageName;
+...
+} SYSTEM_PROCESS_INFORMATION
+```
+
+In C#, we can write it like this:
+
+```csharp
+    internal unsafe struct SYSTEM_PROCESS_INFORMATION
```

Comment:
Code inside of the block should not be indented, it will not look good when rendered.

Reply:
Fixed!